core/tests/test_utils.py

```diff
@@ -196,6 +197,192 @@ def get_storage_model_classes():
                     yield clazz
 
 
+class ElasticSearchServicesStub(python_utils.OBJECT):
+    """The stub class that mocks the API functionality offered by the platform
+    layer, namely the platform.search elastic search services API.
+    """
+
+    _DB = {}
```

Comment:
Please add a line comment explaining the internal structure of this object.

Also, would making it a collection.defaultdict(list) make some of the code simpler (e.g. in add_documents_to_index)?